c2corg_api/views/document_delete.py

```diff
@@ -0,0 +1,327 @@
+        filter(ArchiveDocumentLocale.document_id == document_id). \
+        delete()
+
+
+def _remove_locale(clazz_locale, document_id):
+    document_locale_ids = DBSession.query(DocumentLocale.id). \
+        filter(DocumentLocale.document_id == document_id). \
+        subquery()
+    # Remove links to comments (comments themselves are not removed)
+    DBSession.execute(DocumentTopic.__table__.delete().where(
+        DocumentTopic.document_locale_id.in_(document_locale_ids)
+    ))
+
+    if clazz_locale:
+        DBSession.execute(clazz_locale.__table__.delete().where(
+            getattr(clazz_locale, 'id').in_(document_locale_ids)
+        ))
+
+    DBSession.query(DocumentLocale). \
+        filter(DocumentLocale.document_id == document_id). \
+        delete()
+
+
+def _remove_archive_geometry(document_id):
+    DBSession.query(ArchiveDocumentGeometry). \
+        filter(ArchiveDocumentGeometry.document_id == document_id). \
+        delete()
+
+
+def _remove_geometry(document_id):
+    DBSession.query(DocumentGeometry). \
+        filter(DocumentGeometry.document_id == document_id). \
+        delete()
+
+
+def _remove_archive(archive_clazz, document_id):
+    archive_document_ids = DBSession.query(ArchiveDocument.id). \
+        filter(ArchiveDocument.document_id == document_id). \
+        subquery()
+    DBSession.execute(archive_clazz.__table__.delete().where(
+        getattr(archive_clazz, 'id').in_(archive_document_ids)
+    ))
+
+    DBSession.query(ArchiveDocument). \
+        filter(ArchiveDocument.document_id == document_id). \
+        delete()
+
+
+def _remove_figures(clazz, document_id):
+    DBSession.query(clazz). \
+        filter(getattr(clazz, 'document_id') == document_id). \
+        delete()
+
+
+def _remove_document(document_id):
+    DBSession.query(Document). \
+        filter(Document.document_id == document_id). \
+        delete()
+
+
+def _remove_from_feed(document_id):
+    DBSession.query(DocumentChange). \
+        filter(DocumentChange.document_id == document_id). \
+        delete()
+
+
+def _remove_image_from_feed(document_id):
+    # If removed doc is an image, it might be needed to remove
+    # any reference to this image in feed items
+    items = DBSession.query(DocumentChange). \
+        filter(or_(
+            DocumentChange.image1_id == document_id,
+            DocumentChange.image2_id == document_id,
+            DocumentChange.image3_id == document_id
+        )).all()
+    for item in items:
+        # Shift image references in the feed item
+        if item.image1_id == document_id:
+            item.image1_id = item.image2_id
+            item.image2_id = item.image3_id
+        elif item.image2_id == document_id:
+            item.image2_id = item.image3_id
+        item.image3_id = None
+        item.more_images = False
+        # FIXME what happens if the deleted image was the only image reference
+        # in the feed item => should the item be removed or replaced?
```

Comment:
If the change type is `'added_photos'` it would probably make sense to remove the entire feed entry.